src/persist-txn/src/txn_read.rs

```diff
@@ -380,22 +397,62 @@ impl<T: Timestamp + Lattice + Codec64> TxnsRead<T> {
     }
 }
 
+/// WIP
+#[derive(Debug)]
+pub struct TxnsReadCapability<T: Clone> {
```

Comment:
I played around with this `TxnsReadCapability` a bunch when I was working on this last, and all of the variations felt pretty gross. Definitely possible that it would be a better api to add a `data_subscribe` operation to TxnsRead (instead of any sort of `data_listen_next`) that returns a channel with `(physical upper, logical upper)` pairs. Said another way, this basically makes the timely operator a very thin wrapper around the channel